File: problems/0401 - 0800/0509/0509.py
class Solution:
    def fib(self, n: int) -> int:
        if n < 2:
            return n
        fib_prev = 0
        fib_curr = 1
        for _ in range(2, n + 1):
            fib_curr += fib_prev
            fib_prev = fib_curr - fib_prev
        return fib_curr

File: problems/0401 - 0800/0509/0509.cpp
class Solution {
public:
    int fib(int n) {
        if (n == 0 || n == 1) {
            return n;
        }
        int fib_prev = 0;
        int fib_curr = 1;
        for (int i = 2; i <= n; ++i) {
            fib_curr += fib_prev;
            fib_prev = fib_curr - fib_prev;
        }
        return fib_curr;
    }
};
